frontend/src/views/camp/__tests__/Admin.spec.js

```diff
@@ -0,0 +1,65 @@
+import { render } from '@testing-library/vue'
+import Admin from '../Admin.vue'
+import Vue from 'vue'
+import Vuetify from 'vuetify'
+
+Vue.use(Vuetify)
+
+const renderWithVuetify = (component, options, callback) => {
+  const root = document.createElement('div')
+  root.setAttribute('data-app', 'true')
+
+  return render(
+    component,
+    {
+      container: document.body.appendChild(root),
+      vuetify: new Vuetify(),
+      ...options
+    },
+    callback
+  )
+}
+
+describe('Admin view', () => {
+  it('shows the danger zone when the user has a manager role', async () => {
+    const { getByText } = renderWithVuetify(Admin, {
+      props: {
+        camp: () => ({
+          role: 'manager',
+          materialLists: () => {},
+          _meta: { loading: false }
+        })
+      },
+      routes: [],
+      mocks: {
+        api: { reload: () => Promise.resolve() },
+        $tc: key => key
+      },
+      stubs: ['camp-settings', 'camp-address', 'camp-periods', 'camp-categories', 'camp-material-lists']
+    })
+
+    getByText('components.camp.campDangerzone.title')
```

Comment:
wird hier etwas assertet?

Reply:
Würde ich intuitiver finden, da man die Details der getByText funktion nicht kennen müsste.
Ist aber geschmackssache.
Wie lange ist das Timeout von getByText?

Reply:
Okay, hab ich angepasst. Wenn der Test fehlschlägt ists halt weiterhin wegen dem Error den `getByText` wirft, nicht wegen dem `expect`. Aber liest sich vermutlich wirklich intuitiver.

Das mit dem Timeout hab ich verwechselt, `getBy*` und `queryBy*` haben keins, das Element wird nur einmal gesucht. Wenn man retries will muss man entweder `waitFor` verwenden, oder die dritte Familie von Queries `findBy*` mit `async`/`await`. `waitFor` hat als Standard-Timeout 1 Sekunde, und `findBy*` sind Convenience Funktionen für die Kombination von `waitFor` und `getBy*`, also ebenfalls standardmässig 1 Sekunde.

Siehe auch https://testing-library.com/docs/queries/about/

Reply:
Danke für die Erklärung.
Bei timeouts in "Unittests" werde ich immer hellhörig. Bin schon genug reingelaufen.